benchmarks/meteor/bench.lua

```diff
@@ -0,0 +1,227 @@
+  local function fill(bb, p)
+    bbc = bbc + 1
+    local n = p+2; if bb[n] then bb[n] = false; fill(bb, n) end
+    n = p-2; if bb[n] then bb[n] = false; fill(bb, n) end
+    n = p-9; if bb[n] then bb[n] = false; fill(bb, n) end
+    n = p-11; if bb[n] then bb[n] = false; fill(bb, n) end
+    n = p+9; if bb[n] then bb[n] = false; fill(bb, n) end
+    n = p+11; if bb[n] then bb[n] = false; fill(bb, n) end
+  end
+
+  -- Generate pruned, sliding decision trees.
+  local dtrees = {{}, {}, {}, {}, {}, {}, {}, {}, {}, {}}
+  local rot = { nil, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {} }
+  for k=0,9 do
+    -- Generate 10 initial pieces from line noise. :-)
+    local t = { 60, 62, byte("@BMBIK@KT@GPIKR@IKIKT@GK@KM@BG", k*3+1, k*3+3) }
+    rot[1] = t
+    for i,xy in ipairs(t) do
+      local x = xy%10; local y = (xy-x-60)/10
+      -- Add 11 more variations by rotating and flipping.
+      for j=2,12 do
+	if j == 7 then y = -y else x,y = (x+3*y)/2, (y-x)/2 end
+	rot[j][i] = x+10*y
+      end
+    end
+    for r,v in ipairs(rot) do
+      -- Exploit symmetry and leave out half of the orientations of one piece.
+      -- The selected piece gives the best reduction of the solution space.
+      if k ~= 3 or r%2 == 0 then
+	-- Normalize to origin, add distance, sort by distance from origin.
+	local m = min(v[1], v[2], v[3], v[4], v[5])
+	for i=1,5 do v[i] = dist[v[i]-m] end
+	sort(v)
+	local v2, v3, v4, v5 = v[2]%256, v[3]%256, v[4]%256, v[5]%256
+	-- Slide the piece across 2 rows, prune the tree, check for islands.
+	for j,p in ipairs(slide) do
+	  bb[p] = false
+	  if ok[p+v2] and ok[p+v3] and ok[p+v4] and ok[p+v5] then -- Prune.
+	    for i=p+1,79 do bb[i] = ok[i] end -- Clear remaining board.
+	    bb[p+v2] = false; bb[p+v3] = false -- Add piece.
+	    bb[p+v4] = false; bb[p+v5] = false
+	    bbc = j -- Flood fill and count the filled positions.
+	    if bb[71] then bb[71] = false; fill(bb, 71) end -- Lower left.
+	    if bb[79] then bb[79] = false; fill(bb, 79) end -- Lower right.
+	    local di = 0
+	    if bbc < 22 then bbc = 26
+	    elseif bbc < 26 then -- Island found, locate it, fill from above.
+	      for i=p+2,79 do if bb[i] then di = i-p; break end end
+	      for i=p-9,p-1 do if ok[i] then fill(bb, i) bbc = bbc - 1 end end
+	    end
+	    if bbc == 26 then -- Prune boards with static islands.
+	      local tb = dtrees[j] -- Build decision tree in distance order.
+	      local ta = tb[v2]; if not ta then ta = {}; tb[v2] = ta end
+	      tb = ta[v3]; if not tb then tb = {}; ta[v3] = tb end
+	      ta = tb[v4]; if not ta then ta = {}; tb[v4] = ta; islands[ta] = di
+	      elseif islands[ta] ~= di then islands[ta] = 0 end
+	      ta[v5] = di*10+k -- Leaves hold island check and piece number.
+	    end
+	  end
+	end
+      end
+    end
+  end
+
+  local s = "local u0,u1,u2,u3,u4,u5,u6,u7,u8,u9" -- Piece use flags.
+  for p=0,99 do if ok[p] then s = s..",b"..p end end -- Board cells.
+  s = s.."\n"..[[
+local countinit = ...
```

Comment:
What's the `...` about here?

Reply:
The top level scope of a script is just another function in Lua that also implicitly takes a variable number of arguments. The ... is how you declare and access the vararg like a tuple of those arguments when you assign it to list of variables it get implicitly unpacked.